File: leetcode/PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0){
            return(false);
        }
        String num = Integer.toString(x);
        int i = 0;
        int j = num.length()-1;
        while(i<j){
            if(num.charAt(i)==num.charAt(j)){
                
            }
            else{
                return(false);
            }
            i++;
            j--;
        }
        return(true);
    }
}

File: leetcode/PalindromeNumber.ex
defmodule Solution do
  @spec is_palindrome(x :: integer) :: boolean
  def is_palindrome(x) do
        str_x = Integer.to_string(x)                                           

        if rem(String.length(str_x), 2) do

            str_x === String.reverse(str_x)

        else 
            [head, tail] = String.split(str_x, String.at(str_x, div(String.length(str_x), 2))  )

            head === String.reverse(tail)
        end
  end
      
end
